janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -0,0 +1,411 @@
+                .addClusteringColumn(COLUMN_COLUMN_NAME, DataType.blob())
+                .addColumn(VALUE_COLUMN_NAME, DataType.blob())
+                .withOptions()
+                .compressionOptions(compressionOptions(configuration))
+                .compactionOptions(compactionOptions(configuration))
+                .compactStorage());
+    }
+
+    private static CompressionOptions compressionOptions(final Configuration configuration) {
+        if (!configuration.get(CF_COMPRESSION)) {
+            // No compression
+            return noCompression();
+        }
+
+        return Match(configuration.get(CF_COMPRESSION_TYPE)).of(
+                Case("LZ4Compressor", lz4()),
+                Case("SnappyCompressor", snappy()),
+                Case("DeflateCompressor", deflate()))
+                .withChunkLengthInKb(configuration.get(CF_COMPRESSION_BLOCK_SIZE));
+    }
+
+    private static CompactionOptions<?> compactionOptions(final Configuration configuration) {
+        if (!configuration.has(COMPACTION_STRATEGY)) {
+            return null;
+        }
+
+        final CompactionOptions<?> compactionOptions = Match(configuration.get(COMPACTION_STRATEGY))
+                .of(
+                        Case("SizeTieredCompactionStrategy", sizedTieredStategy()),
+                        Case("DateTieredCompactionStrategy", dateTieredStrategy()),
+                        Case("LeveledCompactionStrategy", leveledStrategy()));
+        Array.of(configuration.get(COMPACTION_OPTIONS))
+                .grouped(2)
+                .forEach(keyValue -> compactionOptions.freeformOption(keyValue.get(0), keyValue.get(1)));
+        return compactionOptions;
+    }
+
+    @Override
+    public void close() throws BackendException {
+        this.closer.run();
+    }
+
+    @Override
+    public String getName() {
+        return this.tableName;
+    }
+
+    @Override
+    public EntryList getSlice(final KeySliceQuery query, final StoreTransaction txh) throws BackendException {
+        final Future<EntryList> result = Future.fromJavaFuture(
+                this.executorService,
+                this.session.executeAsync(this.getSlice.bind()
+                        .setBytes(KEY_BINDING, query.getKey().asByteBuffer())
+                        .setBytes(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
+                        .setBytes(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
+                        .setInt(LIMIT_BINDING, query.getLimit())
+                        .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel())))
+                .map(resultSet -> fromResultSet(resultSet, this.getter));
+        awaitInterruptibly(result);
+        return result.getValue().get().getOrElseThrow(EXCEPTION_MAPPER);
+    }
+
+    @Override
+    public Map<StaticBuffer, EntryList> getSlice(final List<StaticBuffer> keys, final SliceQuery query, final StoreTransaction txh) throws BackendException {
```

Comment:
BackendTransaction.java already knows how to implement a parallelized multi query inside the method [BackendTransaction::edgeStoreMultiQuery()](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/diskstorage/BackendTransaction.java#L296). The effective implementation is the same as that of this method. From what I can see, a store should only advertise that it supports the feature `multiQuery` if the underlying backend truly supports multi queries, and the store invokes that functionality. In this case you are just implementing the parallelization manually at the CQLKeyColumnValueStore level.

It seems like we should just advertise that the CQL store does not support multiQuery and let BackendTransaction handle the multi query parallelization.